lib/value/src/value.rs

```diff
@@ -100,7 +100,7 @@ impl PartialEq<Self> for Value {
             }
             (Value::Integer(a), Value::Integer(b)) => a.eq(b),
             (Value::Object(a), Value::Object(b)) => a.eq(b),
-            (Value::Null, Value::Null) => true,
+            (Self::Null, Self::Null) => true,
```

Comment:
This seems.... odd. Does it only complain about this specific one? If that's the case, I'd rather disable this Clippy lint. I'm fine using `Self` consistently, but it's just confusing to mix the two styles.

Reply:
Any other opinions here? I can see both sides of the argument.

Reply:
```
error: unnecessary structure name repetition
   --> lib/value/src/value.rs:103:14
    |
103 |             (Value::Null, Value::Null) => true,
    |              ^^^^^ help: use the applicable keyword: `Self`
    |
note: the lint level is defined here
   --> lib/value/src/lib.rs:7:5
    |
7   |     clippy::nursery,
    |     ^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::use_self)]` implied by `#[deny(clippy::nursery)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#use_self

error: unnecessary structure name repetition
   --> lib/value/src/value.rs:103:27
    |
103 |             (Value::Null, Value::Null) => true,
    |                           ^^^^^ help: use the applicable keyword: `Self`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#use_self
```

Reply:
I'd be 👍 on using `Self` everywhere here.

Reply:
Updated that and checking to see if it's angry about that at all

Reply:
Yeah, I believe it was only catching the non-tuple, non-struct variant here but probably intends to cover all of them. I'd agree with updating to `Self` everywhere.